core/src/main/scala/com/mesosphere/usi/core/matching/OfferMatcher.scala

```diff
@@ -1,35 +1,80 @@
 package com.mesosphere.usi.core.matching
-
-import com.mesosphere.usi.core.models.{TaskName, RunningPodSpec}
+import com.mesosphere.usi.core.models.template.RunTemplate.KeyedResourceRequirement
+import com.mesosphere.usi.core.models.resources.ResourceType
+import com.mesosphere.usi.core.models.{RunningPodSpec, TaskName}
 import org.apache.mesos.v1.{Protos => Mesos}
 
-/**
-  * Interface for the offer matcher implementations. Given a Mesos [[Mesos.Offer]] and a set of [[RunningPodSpec]]s
-  * return a map of all matched PodSpecs with the list of matched Mesos resources.
-  */
-trait OfferMatcher {
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+
+class OfferMatcher(masterDomainInfo: Mesos.DomainInfo) {
```

Comment:
What was the motivation to move the implementation here?

Reply:
It was premature generalization.